src/styles/main.scss

```diff
@@ -0,0 +1,88 @@
+$color-primary: #060b35;
+$day-color: #eee;
+$border-color: #000;
+$active-color: #ffbfcb;
+$day-size: 100px;
+$font-size: 30px;
+$start-days: mon, tue, wed, thu, fri, sat, sun;
+$month-lengths: 28, 29, 30, 31;
+$default-margin-left: 6 * ($day-size + 1px);
+
+.calendar-wrapper {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  min-height: 100vh;
+}
+
+.calendar {
+  display: flex;
+  align-content: flex-start;
+  flex-wrap: wrap;
+  width: calc($day-size * 7 + 26px);
+  margin: 0 auto;
+  gap: 1px;
+  box-sizing: border-box;
+  padding: 0 10px;
+
+  .calendar__day {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    box-sizing: border-box;
+    height: $day-size;
+    width: $day-size;
+    background-color: $day-color;
+    border: 1px solid $border-color;
+  }
+
+  .calendar__day:hover {
+    background-color: $active-color;
+    transform: translateY(-20px);
+    cursor: pointer;
+    transition:
+      background-color 0.5s,
+      transform 0.5s;
+  }
+
+  .calendar__day:first-child {
+    margin-left: $default-margin-left;
+  }
+
+  @each $day in $start-days {
+    &--start-day-#{$day} &__day:first-child {
```

Comment:
There is a potential issue with the selector `&--start-day-#{$day} &__day:first-child`. The `&` symbol is used to reference the parent selector, but in this context, it seems like you want to target a specific class. Ensure that the intended class structure is correct and that the `&` is used appropriately.